monai/transforms/compose.py

```diff
@@ -308,54 +238,39 @@ class Compose(Randomizable, InvertibleTransform):
             defaults to `True`.
         unpack_items: whether to unpack input `data` with `*` as parameters for the callable function of transform.
             defaults to `False`.
-        log_stats: whether to log the detailed information of data and applied transform when error happened,
-            for NumPy array and PyTorch Tensor, log the data shape and value range,
-            for other metadata, log the values directly. default to `False`.
-        lazy_evaluation: whether to enable lazy evaluation for lazy transforms. If False, transforms will be
-            carried out on a transform by transform basis. If True, all lazy transforms will
+        lazy: whether to enable lazy evaluation for lazy transforms. This can be either string values from the enum
+            ``LazyMode`` or an optional bool. If LazyMode.OFF, lazy execution is disabled and transforms will be
+            carried out on a transform by transform basis. If LazyMode.ON, all lazy transforms will
             be executed by accumulating changes and resampling as few times as possible.
-            A `monai.transforms.Identity[D]` transform in the pipeline will trigger the evaluation of
+            A `monai.transforms.ApplyPending[d]` transform in the pipeline will trigger the evaluation of
             the pending operations and make the primary data up-to-date.
         overrides: this optional parameter allows you to specify a dictionary of parameters that should be overridden
             when executing a pipeline. These each parameter that is compatible with a given transform is then applied
-            to that transform before it is executed. Note that overrides are currently only applied when lazy_evaluation
-            is True. If lazy_evaluation is False they are ignored.
+            to that transform before it is executed. Note that overrides are currently only applied when lazy
+            is True. If lazy is False they are ignored.
             currently supported args are:
             {``"mode"``, ``"padding_mode"``, ``"dtype"``, ``"align_corners"``, ``"resample_mode"``, ``device``},
             please see also :py:func:`monai.transforms.lazy.apply_transforms` for more details.
-        override_keys: this optional parameter specifies the keys to which ``overrides`` are to be applied. If
-            ``overrides`` is set, ``override_keys`` must also be set.
-        verbose: whether to print debugging info when lazy_evaluation=True.
     """
 
     def __init__(
         self,
         transforms: Sequence[Callable] | Callable | None = None,
         map_items: bool = True,
         unpack_items: bool = False,
-        log_stats: bool = False,
-        lazy_evaluation: bool | None = None,
+        lazy: str | bool | None = LazyMode.OFF,
```

Comment:
logically this `Compose.lazy` which as type `str|bool|None` is different from `LazyTransform.lazy` (type `bool|None`) https://github.com/Project-MONAI/MONAI/blob/b16a87fce2ecf51a10efe28ca9eb9bfbf2aa89ba/monai/transforms/transform.py#L281-L284

but they share the same property name `lazy`,  I found it's confusing.

Reply:
This is a consequence of nesting Composes. Compose instances can force non-lazy execution, lazy execution or can leave the choice to the transforms that it contains. bool | none makes sense for containers like Compose, but not for transforms. However, lazy still means lazy. It is just that the semantics are one step more complicated for Compose due to its container nature

Reply:
I don't think it's the same, `Compose` doesn't inherent `LazyTransform` and the outcome of `Compose` always has a `execute_pending_transforms`, which is different from the other subclasses of `LazyTransform`

Reply:
@wyli LazyMode has been replaced by True/False/None